### PR TITLE
Refactor stop_node for safer termination with new tests and automated CI

### PR DESCRIPTION
**Pull request by MaxisAlexander**
## Description

This PR delivers a targeted improvement to node shutdown handling in the **Anoma Supervisor**:

- Refactored `stop_node/1` to use `DynamicSupervisor.terminate_child/2` with a safe fallback to `Supervisor.stop/1`.
- Ensures idempotent behavior and proper handling of unknown or already-stopped nodes.
- Added `supervisor_test.exs` covering normal and edge cases, including process monitoring.
- Introduced a **GitHub Actions CI** workflow for automated build and test execution, including protobuf compilation support.

---

## Impact

- Strengthens reliability of node lifecycle management.
- Provides automated quality checks for future contributions.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+name: CI
```
Comment (mariari) [inline .github/workflows/ci.yml:1]: Very bizzare, we already have CI setup

File: .tool-versions [hunk 2]
```diff
@@ -0,0 +1 @@
+elixir 1.17.3-otp-25
```
Comment (mariari) [inline .tool-versions:1]: We are not adding this, why commit this

File: apps/anoma_node/lib/supervisor.ex [hunk 3]
```diff
@@ -75,9 +75,25 @@ defmodule Anoma.Supervisor do
...
+        case DynamicSupervisor.terminate_child(Anoma.Node.NodeSupervisor, pid) do
+          :ok ->
+            :ok
+
+          # If it's no longer tracked by the DS (race), stop directly and treat as success
+          {:error, :not_found} ->
+            _ = Supervisor.stop(pid)
+            :ok
+        end
+    end
```
Comment (mariari) [inline apps/anoma_node/lib/supervisor.ex:96]: This doesn't make a functional difference.

```elixir
iex(anoma@YU-NO)16> node = Anoma.Node.Examples.ENode.start_node
%Anoma.Node.Examples.ENode{
  pid: #PID<0.982.0>,
  node_id: "D6357773AE9666F46E5ECC81EB3D7ED1907B4ECA30209495DF1C251BEF13893B"
}
iex(anoma@YU-NO)17> GenServer.whereis(Anoma.Node.Registry.via(node.node_id, Anoma.Node.Supervisor))
#PID<0.982.0>
iex(anoma@YU-NO)18> Anoma.Supervisor.stop_node(node.node_id)
:ok
iex(anoma@YU-NO)19> GenServer.whereis(Anoma.Node.Registry.via(node.node_id, Anoma.Node.Supervisor))
nil
```

without the patch included.

If one is trying to stop a node that doesn't exist it's fine for it to do nothing

File: apps/anoma_node/test/supervisor_test.exs [hunk 4]
```diff
@@ -0,0 +1,37 @@
+defmodule Anoma.SupervisorTest do
```
Comment (mariari) [inline apps/anoma_node/test/supervisor_test.exs:1]: We don't write tests in the codebase, see the style guide if you want to contribute here

**Review by mariari — CHANGES_REQUESTED** [review]
Nothing in this PR improves things, I could believe the dynamic supervisor api could be better here but looking at the behaviour I don't think this does anything
